code/trainer.py

```diff
@@ -4,22 +4,23 @@
 import torch.backends.cudnn as cudnn
 import torch
 import torch.nn as nn
-from torch.autograd import Variable
 import torch.optim as optim
 import torchvision.utils as vutils
 import numpy as np
 import os
 import time
+
+from torch.autograd import Variable
 from PIL import Image, ImageFont, ImageDraw
 from copy import deepcopy
 
 from miscc.config import cfg
 from miscc.utils import mkdir_p
+from model import G_NET, D_NET64, D_NET128, D_NET256, D_NET512, D_NET1024, INCEPTION_V3
+
+from tensorboardX import FileWriter, summary
```

Comment:
I would suggest there to be used:
`from torch.utils.tensorboard import summary, FileWriter`